cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -360,7 +360,7 @@
 
   <property>
     <name>app.program.spark.compat</name>
-    <value>spark1_2.10</value>
+    <value>spark2_2.11</value>
```

Comment:
I believe removing it will break the unit tests.

Reply:
We better have this set in the `cdap-site.xml` under `cdap-unit-test/src/main/resources` (and the `cdap-unit-test-spark2_2.11` as well). I don't have a strong opinion on this, so it's ok to keep this as is, just that I feel it is a bit misleading when looking at the `cdap-default.xml`, since changing the value there won't affect anything (or it will affect cdap-unit-test tests? meaning if changing it to spark2, then we don't be running Spark 1 tests?)

Reply:
I agree it's better to have it in the unit test specific resources and that it's confusing have it in cdap-default at all. I'm not sure if we have any unit tests that don't use cdap-unit-test, but which will go through the spark version detection code paths. Though we can just try it and see if anything breaks.